lib/bundlex/toolchain/common/unix/os_deps.ex

```diff
@@ -134,15 +152,20 @@ defmodule Bundlex.Toolchain.Common.Unix.OSDeps do
 
       true ->
         try do
-          File.mkdir(package_path)
-          temporary_destination = "#{@precompiled_path}/temporary"
+          File.mkdir!(package_path)
+          temporary_destination = "#{get_precompiled_path()}/temporary"
           download(precompiled_dependency_url, temporary_destination)
-          System.shell("tar -xf #{temporary_destination} -C #{package_path} --strip-components 1")
-          System.shell("rm #{temporary_destination}")
+
+          {_output, 0} =
+            System.shell(
+              "tar -xf #{temporary_destination} -C #{package_path} --strip-components 1"
+            )
+
+          File.rm!(temporary_destination)
           package_path
         rescue
           e ->
-            Logger.warning("Couldn't download the dependency due to: #{inspect(e)}.")
+            IO.warn("Couldn't download the dependency due to: #{inspect(e)}.")
             :unavailable
         end
```

Comment:
This `try` shouldn't rescue exception from `File.mkdir!/1` function

Reply:
Because there is a possibility that in example, due to network conditions, the dependency won't be downloaded